### PR TITLE
Enables setting of keyword depth_max to None for get_dataset_ids

### LINKED ISSUE #29: Filter depths by depth_from only
@daberer If the option to filter sensors only by their `depth_from` (and ignoring the `depth_to`) really makes sense, we should add some option to https://github.com/TUW-GEO/ismn/blob/9e1a358dc5701da0df144d9b500ca183df92c366/src/ismn/interface.py#L707 to do that.

I would suggest either adding an argument to ignore the `max_depth` or to set a rule (e.g. if `max_depth` is `None` it is ignored)

### PR DESCRIPTION
**Pull request by daberer**
fix #29

### CODE REVIEW

File: src/ismn/interface.py [hunk 1]
```diff
@@ -751,9 +751,9 @@ def get_dataset_ids(self, variable, min_depth=0, max_depth=0.1,
                 * climate_insitu
         """
         lc_cl = ['landcover_2000', 'landcover_2005', 'landcover_2010', 'landcover_insitu', 'climate', 'climate_insitu']
-
-        if max_depth < min_depth:
-            raise ValueError("max_depth can not be less than min_depth")
+        if max_depth:
+            if max_depth < min_depth:
+                raise ValueError("min_depth can not be more than max_depth (default value 0.1)")
```
Comment (wpreimes) [inline src/ismn/interface.py:756]: something like  f"min_depth can not be more than max_depth, min_depth is {min_depth} while max_depth is {max_depth}"

File: src/ismn/interface.py [hunk 2]
```diff
@@ -763,10 +763,15 @@ def get_dataset_ids(self, variable, min_depth=0, max_depth=0.1,
             else:
                 raise ValueError('Specified keyword \"{}\" not found in metadata! Use one of the following: {}'.format(k, lc_cl))
 
-        ids = np.where((self.metadata['variable'] == variable) &
-                       (self.metadata['depth_to'] <= max_depth) &
-                       (self.metadata['depth_from'] >= min_depth) &
-                       landcover_climate)[0]
+        if not max_depth:
```
Comment (wpreimes) [inline src/ismn/interface.py:766]: so now max_depth is optional, but min_depth is mandatory? Fine for me, but should be described in the doc string as well

**Review by wpreimes — CHANGES_REQUESTED** [review]

File: src/ismn/interface.py (hunk 2 above)
Comment (daberer) [reply src/ismn/interface.py:766]: as you suggested, max_depth : float and None, optional (default: 0.1)

File: src/ismn/interface.py (hunk 1 above)
Comment (daberer) [reply src/ismn/interface.py]: "min_depth can not be more than max_depth. min_depth: {}, max_depth: {})".format(min_depth, max_depth)

**Review by wpreimes — APPROVED** [review]
Looks good to me now

**Pull request merged**